src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -0,0 +1,1065 @@
+                        " ch: " + (int)string.charAt(index + i)));
+            }
+        }
+    }
+
+    /**
+     * Expands new line characters to escaped newlines for display.
+     *
+     * @param string a string
+     * @return a string with newline characters escaped
+     */
+    private static String escapeNL(String string) {
+        return string.replace("\n", "\\n")
+                .replace("\r", "\\r");
+    }
+
+    /**
+     * Returns the hexadecimal character for the low 4 bits of the value considering it to be a byte.
+     * If the parameter {@link #isUpperCase()} is {@code true} the
+     * character returned for values {@code 10-15} is uppercase {@code "A-F"},
+     * otherwise the character returned is lowercase {@code "a-f"}.
+     * The values in the range {@code 0-9} are returned as {@code "0-9"}.
+     *
+     * @param value a value, only the low 4 bits {@code 0-3} of the value are used
+     * @return the hexadecimal character for the low 4 bits {@code 0-3} of the value
+     */
+    public char toLowHexDigit(int value) {
+        return (char)digits[value & 0xf];
+    }
+
+    /**
+     * Returns the hexadecimal character for the high 4 bits of the value considering it to be a byte.
+     * If the parameter {@link #isUpperCase()} is {@code true} the
+     * character returned for values {@code 10-15} is uppercase {@code "A-F"},
+     * otherwise the character returned is lowercase {@code "a-f"}.
+     * The values in the range {@code 0-9} are returned as {@code "0-9"}.
+     *
+     * @param value a value, only bits {@code 4-7} of the value are used
+     * @return the hexadecimal character for the bits {@code 4-7} of the value
+     */
+    public char toHighHexDigit(int value) {
+        return (char)digits[(value >> 4) & 0xf];
+    }
+
+    /**
+     * Returns the two hexadecimal characters for the {@code byte} value.
+     * Each nibble (4 bits) from most significant to least significant of the value
+     * is formatted as if by {@link #toLowHexDigit(int) toLowHexDigit(nibble)}.
+     * The delimiter, prefix and suffix are not used.
+     *
+     * @param value a byte value
+     * @return the two hexadecimal characters for the byte value
+     */
+    public String toHexDigits(byte value) {
+        byte[] rep = new byte[2];
+        rep[0] = (byte)toHighHexDigit(value);
+        rep[1] = (byte)toLowHexDigit(value);
+        try {
+            return jla.newStringNoRepl(rep, StandardCharsets.ISO_8859_1);
+        } catch (CharacterCodingException cce) {
+            throw new AssertionError(cce);
+        }
+    }
+
+    /**
+     * Appends two hexadecimal characters for the byte value to the {@link Appendable}.
+     * Each nibble (4 bits) from most significant to least significant of the value
+     * is formatted as if by {@link #toLowHexDigit(int) toLowHexDigit(nibble)}.
+     * The hexadecimal characters are appended in one or more calls to the
+     * {@link Appendable} methods.
+     *
```

Comment:
For parity with the method `public String toHexDigits(byte value)` above I would suggest to add the text

```
     * The delimiter, prefix and suffix are not used.
```

here too.